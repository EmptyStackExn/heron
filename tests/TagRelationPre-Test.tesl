h2 periodic 1.0
tag relation h1 = pre h2
tag relation h0 = [-1.0] -> h1

@policy asap
@step
@step
@step
@step
@step
@step
@print
@output tikz select h2 h1 h0