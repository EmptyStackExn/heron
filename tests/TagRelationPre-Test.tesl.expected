% Exported execution trace
% Solver: Heron 0.53.0-alpha+20181104
% Date: Sun Nov 04 16:16:01 2018

\begin{tikzpicture}[>=stealth,x=\xlength,y=\ylength]
  % - Non-ticking ticks -
  % - Vertical axis -
  \foreach \i in {1,...,6} {
    \draw[edward, vline] (\i+.1,-2) -- +(0, 2.7) ;
  }
  % - Horizontal axis -
  \node at (-0.2,0) (h2) {\texttt{h2}} ;
  \draw[->] (h2.east) -- +(6.5,0) ;
  \node at (-0.2,-1) (h1) {\texttt{h1}} ;
  \draw[->] (h1.east) -- +(6.5,0) ;
  \node at (-0.2,-2) (h0) {\texttt{h0}} ;
  \draw[->] (h0.east) -- +(6.5,0) ;
  % - Ticking ticks and timestamps -
  \node[tick] at (1, 0)  {} ;
  \node[date] at (1, 0)  {0.} ;
  \node[date] at (1, -2)  {-1.} ;
  \node[date] at (2, 0)  {1.} ;
  \node[tick] at (2, 0)  {} ;
  \node[date] at (2, -1)  {0.} ;
  \node[date] at (2, -2)  {0.} ;
  \node[date] at (3, 0)  {2.} ;
  \node[tick] at (3, 0)  {} ;
  \node[date] at (3, -1)  {1.} ;
  \node[date] at (3, -2)  {1.} ;
  \node[date] at (4, 0)  {3.} ;
  \node[tick] at (4, 0)  {} ;
  \node[date] at (4, -1)  {2.} ;
  \node[date] at (4, -2)  {2.} ;
  \node[date] at (5, 0)  {4.} ;
  \node[tick] at (5, 0)  {} ;
  \node[date] at (5, -1)  {3.} ;
  \node[date] at (5, -2)  {3.} ;
  \node[date] at (6, 0)  {5.} ;
  \node[tick] at (6, 0)  {} ;
  \node[date] at (6, -1)  {4.} ;
  \node[date] at (6, -2)  {4.} ;
  % - Instant indexes -
  \foreach \i/\x in {1/1,2/2,3/3,4/4,5/5,6/6} {
    \node (I\i) at (\x+0.1, -2) {} ; 
    \node[instantlab] at (I\i.south) {\textsf{\i}} ;
  }
\end{tikzpicture}
